src/main/java/com/gorlah/kappabot/function/response/formatter/MarkdownToHtmlResponseFormatter.java

```diff
@@ -0,0 +1,27 @@
+package com.gorlah.kappabot.function.response.formatter;
+
+import com.gorlah.kappabot.function.response.BotResponse;
+import org.commonmark.parser.Parser;
+import org.commonmark.renderer.html.HtmlRenderer;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+class MarkdownToHtmlResponseFormatter implements ResponseFormatter {
+
+    private final Parser parser = Parser.builder().build();
+    private final HtmlRenderer renderer = HtmlRenderer.builder().build();
+
+    @Override
+    public Optional<String> format(BotResponse response, String contentType) {
+        return Stream.of(response)
+                .filter(resp -> resp.getContentType().equals("text/markdown"))
+                .filter(resp -> contentType.equals("text/html"))
```

Comment:
I feel like these really need to be enums.

Reply:
They could probably be constants at least but I think using an enum would detract from the flexibility of the interface. I decided to go with a String so that for example if KappaBot supported loading plugins in the future, `ResponseFormatter`s and `CommandAdapter`s would be able to handle a native format of `BotResponse` that is not accounted for in the main KappaBot code.

For example, say you had a plugin that provides a `CommandAdapter` that declares a contentType of "application/vnd-kb-plugin1" and a `BotFunction` that produces `BotResponse`s in that format. Let's say that "application/vnd-kb-plugin1" is some kind of enhanced formatting message type, something besides plain text or markdown. Then if you also wanted to provide a `ResponseFormatter` that can convert "application/vnd-kb-plugin1" to "text/plain" or "text/markdown" (so that Discord and anything else can understand your `BotFunction`'s special message), I don't think that would be possible with using an enum unless you went back and changed the code in the main project.

So the rationale for using a String here is flexibility relative to an enum. Using something like `MediaType` would be another option. If you still feel strongly about not using a String here we can look at other options.

Reply:
For now I extracted constants for the Strings.

Reply:
That definitely makes sense. My only concern with this implementation was how easy it would be for a typo to slip through unnoticed. I guess the only other option that would allow for extension would be making a MediaType interface, but that is probably overkill at that point.

The string constants should do for now.